json_serializable/lib/src/helper_core.dart

```diff
@@ -75,7 +75,11 @@ $converterOrKeyInstructions
 * Set `JsonSerializable.genericArgumentFactories` to `true`
   https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonSerializable/genericArgumentFactories.html''';
   } else if (field.type != error.type) {
-    message = '$message because of type `${typeToCode(error.type)}`';
+    try {
+      message = '$message because of type `${typeToCode(error.type)}`';
```

Comment:
We should avoid doing a catch without details. What type is thrown? Unimplemented? From which type?

Reply:
Yea, specifically this line is throwing the error

`package:json_serializable/src/utils.dart 221:3`
```
throw UnimplementedError('(${type.runtimeType}) $type');
```

For me `type.runtimeType` is equal to `InvalidTypeImpl` when printed in the error.

Would you prefer a refactor like this:

```
try {
      message = '$message because of type `${typeToCode(error.type)}`';
} on UnimplementedError catch (ex) {
     message = '$message because type is Unimplemented ($ex)';
}
```